### PR TITLE
[BEAM-4374] Emit SampledByteCount distribution tuple system metric from Python SDK (@Ardagan helped contribute)

### PR DESCRIPTION
**Pull request by ajamato**
[BEAM-4374] Emit SampledByteCount distribution tuple system metric from Python SDK

Thank you to @Ardagan who contributed to this PR as well.

------------------------

Thank you for your contribution! Follow this checklist to help us incorporate your contribution quickly and easily:

 - [ ] [**Choose reviewer(s)**](https://beam.apache.org/contribute/#make-your-change) and mention them in a comment (`R: @username`).
 - [ ] Format the pull request title like `[BEAM-XXX] Fixes bug in ApproximateQuantiles`, where you replace `BEAM-XXX` with the appropriate JIRA issue, if applicable. This will automatically link the pull request to the issue.
 - [ ] If this contribution is large, please file an Apache [Individual Contributor License Agreement](https://www.apache.org/licenses/icla.pdf).

Post-Commit Tests Status (on master branch)
------------------------------------------------------------------------------------------------

Lang | SDK | Apex | Dataflow | Flink | Gearpump | Samza | Spark
--- | --- | --- | --- | --- | --- | --- | ---
Go | [Build Status](https://builds.apache.org/job/beam_PostCommit_Go/lastCompletedBuild/) | --- | --- | --- | --- | --- | ---
Java | [Build Status](https://builds.apache.org/job/beam_PostCommit_Java/lastCompletedBuild/) | [Build Status](https://builds.apache.org/job/beam_PostCommit_Java_ValidatesRunner_Apex/lastCompletedBuild/) | [Build Status](https://builds.apache.org/job/beam_PostCommit_Java_ValidatesRunner_Dataflow/lastCompletedBuild/) | [Build Status](https://builds.apache.org/job/beam_PostCommit_Java_ValidatesRunner_Flink/lastCompletedBuild/)<br>[Build Status](https://builds.apache.org/job/beam_PostCommit_Java_PVR_Flink_Batch/lastCompletedBuild/)<br>[Build Status](https://builds.apache.org/job/beam_PostCommit_Java_PVR_Flink_Streaming/lastCompletedBuild/) | [Build Status](https://builds.apache.org/job/beam_PostCommit_Java_ValidatesRunner_Gearpump/lastCompletedBuild/) | [Build Status](https://builds.apache.org/job/beam_PostCommit_Java_ValidatesRunner_Samza/lastCompletedBuild/) | [Build Status](https://builds.apache.org/job/beam_PostCommit_Java_ValidatesRunner_Spark/lastCompletedBuild/)
Python | [Build Status](https://builds.apache.org/job/beam_PostCommit_Python_Verify/lastCompletedBuild/)<br>[Build Status](https://builds.apache.org/job/beam_PostCommit_Python3_Verify/lastCompletedBuild/) | --- | [Build Status](https://builds.apache.org/job/beam_PostCommit_Py_VR_Dataflow/lastCompletedBuild/) <br> [Build Status](https://builds.apache.org/job/beam_PostCommit_Py_ValCont/lastCompletedBuild/) | [Build Status](https://builds.apache.org/job/beam_PreCommit_Python_PVR_Flink_Cron/lastCompletedBuild/) | --- | --- | ---

See [.test-infra/jenkins/README](https://github.com/apache/beam/blob/master/.test-infra/jenkins/README.md) for trigger phrase, status and link of all Jenkins jobs.

### CODE REVIEW

**Comment by ajamato** [comment]
@pabloem @Ardagan @ryan-williams

**Comment by pabloem** [comment]
Hm the tests have failed - I'll retrigger them, but if they still fail, I'll wait for you alex.

**Comment by pabloem** [comment]
Run Python PreCommit

**Comment by Ardagan** [comment]
Just a heads up @ajamato 
I forked this PR and fixing issues here: https://github.com/apache/beam/pull/8136
We can figure out how to follow up on this once Alex is back.

**Comment by ajamato** [comment]
@Ardagan I reset my branch and pushed it back to this PR, I'll continue iterating on this and fixing it up. Thanks for the contributions

**Comment by ajamato** [comment]
Run Java PreCommit

**Comment by ajamato** [comment]
Run Python PreCommit

**Comment by ajamato** [comment]
Run Python PreCommit

File: sdks/python/apache_beam/runners/portability/fn_api_runner_test.py [hunk 1]
```diff
@@ -611,35 +691,99 @@ def process(self, element):
...
-    # outputs.
-
-    assert_contains_metric(counters, monitoring_infos.ELEMENT_COUNT_URN,
-                           'ref_PCollection_PCollection_5', 8)
-    assert_contains_metric(counters, monitoring_infos.ELEMENT_COUNT_URN,
-                           'ref_PCollection_PCollection_6', 8)
-    assert_contains_metric(counters, monitoring_infos.ELEMENT_COUNT_URN,
-                           'ref_PCollection_PCollection_7', 2)
+      # Create/Read, "out" output.
+      labels = {'PCOLLECTION' : 'ref_PCollection_PCollection_1'}
```
Comment (Ardagan) [inline sdks/python/apache_beam/runners/portability/fn_api_runner_test.py]: use constant for monitoring_infos.PCOLLECTION_LABEL

**Comment by pabloem** [comment]
Run Java PreCommit

**Comment by pabloem** [comment]
Run Python PreCommit

**Comment by ajamato** [comment]
Run Python PreCommit

**Comment by ajamato** [comment]
Run Java_Examples_Dataflow PreCommit

File: sdks/python/apache_beam/runners/worker/bundle_processor.py [hunk 2]
```diff
@@ -675,14 +675,15 @@ def monitoring_infos(self):
...
-    def inject_pcollection_into_element_count(monitoring_info):
+    def inject_pcollection(monitoring_info):
       """
       If provided metric is element count metric:
       Finds relevant transform output info in current process_bundle_descriptor
       and adds tag with PCOLLECTION_LABEL and pcollection_id into monitoring
       info.
       """
-      if monitoring_info.urn == monitoring_infos.ELEMENT_COUNT_URN:
+      if monitoring_info.urn in set([monitoring_infos.ELEMENT_COUNT_URN,
```
Comment (Ardagan) [inline sdks/python/apache_beam/runners/worker/bundle_processor.py]: This set can be extracted to final field.

**Comment by ajamato** [comment]
Run Java PreCommit

**Comment by ajamato** [comment]
Run Java PreCommit

**Comment by ajamato** [comment]
Run Java PreCommit

**Comment by ajamato** [comment]
Run Java PreCommit

**Comment by pabloem** [comment]
Run Python PostCommit

**Pull request merged**